### ISSUE TITLE
[BUG] Backlash Compensation Skipping Steps

### ISSUE THREAD

**Issue by stabbs** [MarlinFirmware/Marlin#19478]
I just updated my printer's firmware to version 2.0.6.1 to take advantage of backlash compensation on the z axis.  I have a worm / belt driven z bed and there is 1.94mm of backlash measured with a dial indicator.  I am also using x,y, and z extruder offset for tool changes.

  #define BACKLASH_DISTANCE_MM { 0, 0, 1.94 } // (mm)
  #define BACKLASH_CORRECTION    1.0       // 0.0 = no correction; 1.0 = full correction

The z stepper motor sounds like its trying to spin faster depending on the reversing distance.  I posted a YouTube link below showing this issue as well.  For example:
G1 Z1.0 F100
G1 Z0.0 F100    -  Backlash compensation works well, no issues

G1 Z1.0 F100
G1 Z0.9 F100   -  Motors strain and skip some steps

G1 Z1.0 F100
G1 Z0.99        -  Motors strain and skips all steps

I have my z speed limited to 5mm/s and acceleration at 120mm/s but the backlash compensation ignores these limits.  I have also tried enabling BACKLASH_SMOOTHING_MM 3 but I need the backlash compensation to be immediate for successful tool changes.

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/5262160/Configuration.zip)

Please see this video demonstrating the backlash problem
https://youtu.be/MUfKBlZGgww

Thanks in advance for the help and apologies if this is posted in the wrong place

**Comment by boelle** [comment]
Please test the [`bugfix-2.0.x branch`](https://github.com/MarlinFirmware/Marlin/tree/bugfix-2.0.x) to see where it stands.

**Comment by stabbs** [comment]
Thank you for your reply.  I'm getting the following error: 

Error compiling for board Arduino Mega or Mega 2560
avr-gcc: error: CreateProcess: No such file or directory
exit status 1
Error compiling for board Arduino Mega or Mega 2560.


Followed[ the instructions below](https://forum.arduino.cc/index.php?topic=621875.0) to fix the compile error but the error persists

Here's the fix:
1.	Tools > Board > Boards Manager
2.	Wait for downloads to finish.
3.	From the list of boards platforms, click "Arduino AVR Boards".
4.	Click the "Update" button.
5.	Wait for installation to finish.
6.	Click "Close".

Edit: Using Arduino 1.9.0 beta to compile

**Comment by sjasonsmith** [comment]
That happens due to command length limitations on Windows that Arduino doesn’t handle well. There are ways to work around it, but it is better to learn how to build Marlin using Visual Studio code and PlatformIO.

**Comment by stabbs** [comment]
**bugfix-2.0.x branch** did not fix or change the backlash compensation issue posted above

Sidenote: I couldn't get **bugfix-2.0.x branch** to compile in Arduino IDE 1.9.0 beta but was able to compile and upload using  PlatformIO.  Also compile times were 16 minutes in Arduino IDE and only 3 in Platform IO

**Comment by sjasonsmith** [comment]
If you disable backlash compensation do those small moves work properly?

From the video it definitely sounds like acceleration is not being applied correctly for small moves, I just want to very whether backlash compensation is actually an important factor in reproducing this.

**Comment by stabbs** [comment]
No issues making small moves with backlash compensation disabled- either when switching direction or going in one direction.  With backlash compensation enabled I have no issues making small moves in one direction, only when reversing direction.

Just for fun I added the same (1.94mm) backlash to the x and y axis to see if they would skip and neither did but the short z moves still skip.

I tried a few other things that didn't work:
- Reduced Backlash_Correction to .5 and made the backlash amount 1.94*2
- Reduced Backlash_Correction to .1 and made backlash amount 1.94*10
- Installed CURA but had the same issue
- Reduced acceleration speeds for x, y and z
- Reduced x,y,and z axis max feedrate

I didn't mention earlier and not sure it's relevant but I'm using a Rambo 1.2d board

**Comment by stabbs** [comment]
I think this issue is related to the steps/mm.  My current setup uses a worm gear with a 30:1 reduction so my z-steps is set to 2445.  I reduced this to 1000 and the backlash compensation seems to work fine- no skipping or whirring.  It starts buzzing/skipping around 1800 steps

**edit:**  I added the 1.94mm backlash to x axis and changed the x esteps, acceleration and feedrate to match the z motor.  It did not skip but when taking up the backlash the motor moves extremely fast when reversing direction by .01mm.  I can measure the exact speed but it seems to exceed the max feedrate which I had set to 5mm/s
